Hazel/Dtls/DtlsConnectionListener.cs

```diff
@@ -965,20 +949,32 @@ private bool HandleClientHello(PeerData peer, IPEndPoint peerAddress, ref Record
 
             Handshake serverHelloHandshake = new Handshake();
             serverHelloHandshake.MessageType = HandshakeType.ServerHello;
-            serverHelloHandshake.Length = ServerHello.Size;
+            serverHelloHandshake.Length = ServerHello.MinSize;
             serverHelloHandshake.MessageSequence = 1;
             serverHelloHandshake.FragmentOffset = 0;
             serverHelloHandshake.FragmentLength = serverHelloHandshake.Length;
 
+            int maxCertFragmentSize = peer.Session.Version == 0 ? MaxCertFragmentSizeV0 : MaxCertFragmentSizeV1;
+
+            // The first certificate data needs to leave room for
+            //  * Record header
+            //  * ServerHello header
+            //  * ServerHello payload
+            //  * Certificate header
+
+            var certificateData = this.encodedCertificate;
+            const int InitialCertPadding = Record.Size + Handshake.Size + ServerHello.MinSize + Handshake.Size;
```

Comment:
Oh wow, I didn't realize you can declare const like this in C# - I'm probably going to start doing that a lot 🔥